### PR TITLE
Default container

### LINKED ISSUE #359: Provide default template parameter to Table::SampleRows
Most applications will use this with `std::vector` so we should make it the default template parameter.

### PR DESCRIPTION
**Pull request by paprat**
This fixed #359.

### CODE REVIEW

File: bigtable/client/table_sample_row_keys_test.cc [hunk 1]
```diff
@@ -51,6 +79,8 @@ TEST_F(TableSampleRowKeysTest, SimpleVectorTest) {
   EXPECT_CALL(*reader, Finish()).WillOnce(Return(grpc::Status::OK));
   auto result = table_.SampleRows<std::vector>();
   auto it = result.begin();
+  EXPECT_EQ(typeid(std::vector<bigtable::v0::Table::RowKeySample>),
```
Comment (coryan) [inline bigtable/client/table_sample_row_keys_test.cc]: Or you could say 

```C++
std::vector<bigtable::Table::RowKeySample> result = ...
```

which would detect the problem at compile time.

File: bigtable/client/table_sample_row_keys_test.cc [hunk 2]
```diff
@@ -76,6 +106,8 @@ TEST_F(TableSampleRowKeysTest, SimpleListTest) {
   EXPECT_CALL(*reader, Finish()).WillOnce(Return(grpc::Status::OK));
   auto result = table_.SampleRows<std::list>();
   auto it = result.begin();
+  EXPECT_EQ(typeid(std::list<bigtable::v0::Table::RowKeySample>),
```
Comment (coryan) [inline bigtable/client/table_sample_row_keys_test.cc]: Ditto.

File: bigtable/client/table_sample_row_keys_test.cc (hunk 1 above)
Comment (paprat) [reply bigtable/client/table_sample_row_keys_test.cc]: Done.

File: bigtable/client/table_sample_row_keys_test.cc (hunk 2 above)
Comment (paprat) [reply bigtable/client/table_sample_row_keys_test.cc]: Done.

**Pull request merged**